gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaAvroJobStatusMonitor.java

```diff
@@ -64,6 +72,8 @@
   @Getter
   private Meter messageParseFailures;
 
+  private final HashMap<String, Long> flowNameGroupToWorkUnitCount;
```

Comment:
I see that other fields in this class are ThreadLocal, which may indicate that this class can be called from multiple threads. Also, I'm pretty sure that the lambda " () -> this.flowNameGroupToWorkUnitCount.get(workUnitCountName)" below will be called by metrics library from a separate thread. 

If a collection is accessed by multiple threads at the same time, you'll get occasional data corruption and strange exceptions.

You can use a thread-safe collection here instead from Apache commons library or Guava.